### PR TITLE
Convert to DTO

### PR DESCRIPTION
**Pull request by arufanov**
Convert to DTO

### CODE REVIEW

**Comment by arufanov** [comment]
@yurichervoniy let discuss:
* how we work with acronym: API or Api, DTO or Dto. What the name of DTO objects: RecordDTO or RecordDto. My opinion DTO and API.
* where we should map: entity to dto: controller level or service level?  I have met the both way. My opinion: repository should work with entiry, controller with dto, service layer is mapper. For example here https://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application is opposite opinion: "Now let's look at a service level operation, which will obviously work with the Entity (not the DTO)" Here is more complex discussion https://stackoverflow.com/questions/20092440/pass-dto-to-service-layer  "Not only you could pass DTO objects to Service Layer, but you should pass DTO objects instead of Business Entities to Service Layer."

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantDTO.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.a5lab.tabr.domain.tenants;
+
+public record TenantDTO(Long id, String title, String description) {
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/TenantDTO.java:3]: Whats changed? =) 
I still see record

**Comment by yurichervoniy** [comment]
> * Dto

The simplest way - use CamelCase
Dto and Api in class names

**Comment by arufanov** [comment]
@yurichervoniy lets discuss:
"tabr/src/main/java/com/a5lab/tabr/domain/tenants/TenantMapper.java:[9,8] Unmapped target property: "id". Occured at 'E toEntity(R record)' in 'EntityToDtoMapper'"

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantDto.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  @Size(min=1, max=512)
+  private String description;
+
+  public TenantDto() {
+    // TODO: remove after add default null for next contructor
+  }
+
+  public TenantDto(String title, String description) {
+    // TODO: where is init list?
+    // TODO: how to make default value
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/TenantDto.java:29]: For default values you can use a bulder pattern. 
A prefer static factory method to create objects - no public constructor (or just builder)

For example
TenantDto e = new TenantDto() - remove such calls
Instead use
TenantDto e = TenantDto.of(...) or TenantDto.create(supply, provide, costruct and so on)

Default values in methods can be used in Kotlin =)
Or you can use methods this differents input values 
method() {
 method("value1", "value2");
}

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantDto.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  @NotBlank
+  @Size(min=1, max=512)
+  private String description;
+
+  public TenantDto() {
+    // TODO: remove after add default null for next contructor
+  }
+
+  public TenantDto(String title, String description) {
+    // TODO: where is init list?
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/TenantDto.java:28]: I don't understand about init list

File: src/main/java/com/a5lab/tabr/domain/tenants/Tenant.java [hunk 4]
```diff
@@ -29,14 +30,15 @@ public class Tenant extends AbstractAuditable {
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "id", nullable = false, updatable = false, unique = true)
-  @Setter(AccessLevel.NONE)
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/Tenant.java:32]: Do we need a setter for ID? 
It's autogenerated, isn't it?

**Comment by arufanov** [comment]
@yurichervoniy 
* lets discuss.

**Review by yurichervoniy — APPROVED** [review]

**Pull request merged**